org.openhab.binding.zigbee/src/main/java/org/openhab/binding/zigbee/internal/converter/config/ZclWarningDeviceConfig.java

```diff
@@ -0,0 +1,73 @@
+/**
+ * Copyright (c) 2010-2018 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.zigbee.internal.converter.config;
+
+import java.math.BigDecimal;
+import java.math.BigInteger;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Objects;
+
+import org.eclipse.jdt.annotation.NonNull;
+import org.eclipse.smarthome.config.core.Configuration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.zsmartsystems.zigbee.zcl.clusters.ZclIasWdCluster;
+
+/**
+ * Configuration handler for the IAS WD (Warning Device) cluster.
+ *
+ * @author Henning Sudbrock - initial contribution
+ */
+public class ZclWarningDeviceConfig {
```

Comment:
I had thought about this, but decided against this as the `ZclClusterConfigHandler` interface is (at least currently) being used for configuration on the thing level (from within the `ZigBeeThingHandler`), whereas the `ZclWarningDeviceConfig` I implemented here is being used only from within a single channel converter for managing channel configuration.

Does this sound sensible?

Reply:
I'm OK with it

Reply:
I'm not sure there's such a distinction is there? The polling config handler works on the channel IIRC. If you think we should have different interfaces for the channel and thing config, I guess we can look at that, but I'm not sure why it's needed?

Reply:
I don't think that we need different interfaces for the channel and the thing config.

I merely do not see the need for `ZclWarningDeviceConfig` implementing the `ZclClusterConfigHandler` interface (or any other interface). (It would need to implement the `ZclClusterConfigHandler` interface if the configuration were at the thing level, so that the thing handler could properly use the class - but that's not the case here.)

Reply:
Sorry - I missed this.

I guess if this config class is only used in one class, and doesn't implement any system interface, then why not just keep the config handling in the converter itself?

Reply:
Hm, valid question. I used a separate class, since I think that it is often useful to separate different concerns into different classes - but I am open to keep it within the converter class. Let me play around with this quickly, to see if it bloats the converter class too much.

Reply:
I think you are right, @cdjackson , here it is ok to put the configuration logic into the converter class itself. I have pushed a commit changing this.